filebeat/module/elasticsearch/slowlog/ingest/pipeline-json-8.yml

```diff
@@ -0,0 +1,36 @@
+description: Pipeline for parsing the Elasticsearch slow logs in JSON format.
+on_failure:
+- set:
+    field: error.message
+    value: '{{ _ingest.on_failure_message }}'
+processors:
+- json:
+    field: message
+    add_to_root: true
+- dot_expander:
+    field: '*'
+    override: true
+- convert:
+    field: elasticsearch.slowlog.took_millis
+    type: float
+    ignore_missing: true
+- rename:
+    field: elasticsearch.slowlog.took_millis
+    target_field: elasticsearch.slowlog.duration
+    ignore_missing: true
+- grok:
+    field: elasticsearch.slowlog.message
```

Comment:
As a follow up, I'm wondering why ES in 8.0 not puts this out as JSON already?

Reply:
This is probably also a question for @pgomulka to help out with. Looks like the `message` field is basically structured, so would be nice to split it out into the json.

Reply:
in ESv8 JSON slow log's message field only contains shardId.
Take a look at formatted log line
```
{
  "@timestamp": "2022-01-27T11:42:31.395Z",
  "log.level": "DEBUG",
  "elasticsearch.slowlog.id": null,
  "elasticsearch.slowlog.message": "[test_1][0]",
  "elasticsearch.slowlog.search_type": "QUERY_THEN_FETCH",
  "elasticsearch.slowlog.source": "{}",
  "elasticsearch.slowlog.stats": "[]",
  "elasticsearch.slowlog.took": "115.3micros",
  "elasticsearch.slowlog.took_millis": 0,
  "elasticsearch.slowlog.total_hits": "2 hits",
  "elasticsearch.slowlog.total_shards": 1,
  "ecs.version": "1.2.0",
  "service.name": "ES_ECS",
  "event.dataset": "elasticsearch.index_search_slowlog",
  "process.thread.name": "elasticsearch[runTask-0][search][T#5]",
  "log.logger": "index.search.slowlog.query",
  "elasticsearch.cluster.uuid": "5alW33KLT16Lp1SevDqDSQ",
  "elasticsearch.node.id": "tVLnAGLgQum5ca6z50aqbw",
  "elasticsearch.node.name": "runTask-0",
  "elasticsearch.cluster.name": "runTask"
}
```
I guess having that shardId in its own field would be nice. Is this what you mean?

Reply:
Looking at the grok pattern used below, it seems there could be much more then just the shardId. But if it only contains the shard id, then the massive grok pattern below becomes obsolete.

Reply:
could it be because the message field in plaintext logs (7.x and 6.x .log files) contained all emitted as structured plaintext. So effectively all fields into message field? 
It looks to me that this pattern could parse it (not really needed because JSON logs will be emitted into .json file - not .log)
https://github.com/elastic/beats/blob/master/filebeat/module/elasticsearch/slowlog/test/test.log

Reply:
Assuming this does not exist anymore in 8.0 and as this is only the pipeline for 8.0 logs, seems like this could be cleaned up then. @matschaffer